src/Components/test/E2ETest/Tests/WebAssemblyICUShardingTest.cs

```diff
@@ -0,0 +1,124 @@
+    public class WebAssemblyICUShardingTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        private readonly DateTime DisplayTime = new DateTime(2020, 09, 02);
+        public WebAssemblyICUShardingTest(
+            BrowserFixture browserFixture,
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+        }
+
+        [Fact]
+        public void LoadingApp_FrenchLanguage_Works()
+        {
+            // Arrange
+            // This verifies the EFIGS icu data set.
+            var culture = new CultureInfo("fr-FR");
+            Initialize(culture);
+
+            var cultureDisplay = Browser.Exists(By.Id("culture"));
+            Assert.Equal(culture.ToString(), cultureDisplay.Text);
+
+            var dateDisplay = Browser.Exists(By.Id("dateTime"));
+            Assert.Equal(DisplayTime.ToString(culture), dateDisplay.Text);
+
+            var localizedDisplay = Browser.Exists(By.Id("localizedString"));
+            Assert.Equal("Bonjour!", localizedDisplay.Text);
+        }
+
+        [Fact]
+        public void LoadingApp_KoreanLanguage_Works()
+        {
+            // Arrange
+            // This verifies the CJK icu data set.
+            var culture = new CultureInfo("ko-KO");
+            Initialize(culture);
+
+            var cultureDisplay = Browser.Exists(By.Id("culture"));
+            Assert.Equal(culture.ToString(), cultureDisplay.Text);
+
+            var dateDisplay = Browser.Exists(By.Id("dateTime"));
+            Assert.Equal(DisplayTime.ToString(culture), dateDisplay.Text);
+
+            var localizedDisplay = Browser.Exists(By.Id("localizedString"));
+            // The app has a "ko" resx file. This test verifies that we can walk up the culture hierarchy correctly.
+            Assert.Equal("안녕하세요", localizedDisplay.Text);
+        }
+
+        [Fact]
+        public void LoadingApp_RussianLanguage_Works()
+        {
+            // Arrange
+            // This verifies the non-CJK icu data set.
+            var culture = new CultureInfo("ru");
+            Initialize(culture);
+
+            var cultureDisplay = Browser.Exists(By.Id("culture"));
+            Assert.Equal(culture.ToString(), cultureDisplay.Text);
+
+            var dateDisplay = Browser.Exists(By.Id("dateTime"));
+            Assert.Equal(DisplayTime.ToString(culture), dateDisplay.Text);
+
+            var localizedDisplay = Browser.Exists(By.Id("localizedString"));
+            Assert.Equal("Hello", localizedDisplay.Text); // No localized resources for this culture.
+        }
+
+        [Fact(Skip = "Figure out why this is broken")]
+        public void LoadingApp_KannadaLanguage_Works()
+        {
+            // Arrange
+            // This verifies the non-CJK icu data set.
+            var culture = new CultureInfo("kn");
```

Comment:
@lewing \ @EgorBo I authored a few tests to verify that Blazor is loading icu resources correctly one for each of the 3 shards. I verified that we're loading the right shard (and there's another test that verifies this for the `ru` culture), but this one appears to be failing. 

```
    Assert.Equal() Failure
               ↓ (pos 1)
    Expected: 2/9/2020 12:00:00 ಪೂರ್ವಾಹ್ನ
    Actual:   2020-09-02 00:00:00
```

Reply:
@pranavkm to be clear: is `kn` not expected to work in non-CJK icudt?

Reply:
@EgorBo I would have expected it to work. This doesn't seem specific to the shards. Even if I load `icudt.dat`, there's a bunch of different cultures for which it looks like culture data is missing.

Reply:
@pranavkm sorry for the delay, the fix is here https://github.com/dotnet/icu/pull/34
I guess we need to backport it to 5.0rc2 ?

Reply:
the bug affects the following locales: "bn", "et", "gu", "kn", "ml", "mr", "ms", "no", "ta", "te"